grammars/css.cson

```diff
@@ -500,18 +500,24 @@
             'name': 'punctuation.definition.entity.css'
         'match': '''(?x)
           (\\.)
-          -?+                   # possessive to avoid matching '-' followed by <0-9>
+          -?+                   # possessive to avoid matching '-' followed by [0-9]
           [\\w-&&[^0-9]]        # Do NOT change 0-9 to \d, they are not the same.
           [\\w-]*+
-          (?=$|[\\s,.:{>+~|])   # necessary to avoid matching, e.g., '.B' in '.B&W'
+          (?=$|[\\s,.:{>+~|])   # to avoid matching '.B' in '.B&W'
         '''
         'name': 'entity.other.attribute-name.class.css'
       }
       {
         'captures':
           '1':
             'name': 'punctuation.definition.entity.css'
-        'match': '(#)[a-zA-Z][a-zA-Z0-9_-]*'
+        'match': '''(?x)
+          (\\#)
```

Comment:
It's true that `#` usually doesn't need escaping. But in extended mode (`?x`), `#` becomes a metacharacter, denoting the start of a comment—if you don't escape it, the following `)` will be interpreted as comment.